_enumerable.rb

```diff
@@ -0,0 +1,84 @@
+module Enumerable
+  def my_each
+    each do |ele|
+      yield ele
+    end
+    self
+  end
+
+  def my_each_with_index
+    length.times do |indx|
+      yield self[indx], indx
+    end
+    self
+  end
+
+  def my_select
+    selected_elements = []
+    my_each do |k|
+      selected_elements.push(k) if yield k
+    end
+    selected_elements
+  end
+
+  def my_all?
+    my_each do |ele|
+      # returns false if an element that doesn't satsify the condition is found
+      return false unless yield ele
+    end
+    true
+  end
```

Comment:
#my_all? when no block or argument is given returns true when none of the collection members are false or nil

#my_all? when no block or argument is given returns false when one of the collection members are false or nil

#my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Integer

Is expected that: 
#my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric

#my_all? when a class is passed as an argument returns false if any of the collection is not a member of such class

#my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex

#my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex

#my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

#my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern